core/src/main/java/org/apache/iceberg/expressions/ExpressionParser.java

```diff
@@ -159,8 +162,13 @@ public <T> Void predicate(BoundPredicate<T> pred) {
 
             if (pred.isLiteralPredicate()) {
               gen.writeFieldName(VALUE);
-              SingleValueParser.toJson(
-                  pred.term().type(), pred.asLiteralPredicate().literal().value(), gen);
+              if (pred.op() == Operation.ST_INTERSECTS || pred.op() == Operation.ST_DISJOINT) {
```

Comment:
since bounding box literal is a byte buffer type, it can be json serialized as as binary types. I guess the special handling here is to make the json serialization more human readable.